exec/plugin-commands-script-runners/src/shouldRunCheck.ts

```diff
@@ -1,4 +1,5 @@
-// The scripts that `pnpm run` executes are likely to also execute other `pnpm run`.
+// The scripts that `pnpm run` and `pnpm install` execute
+// are likely to also execute other `pnpm run`.
 // We don't want this potentially expensive check to repeat.
 // The solution is to use an env key to disable the check.
 export const SKIP_ENV_KEY = 'pnpm_run_skip_deps_check'
```

Comment:
Maybe we could avoid the additional env variable by setting this instead: 

```
npm_config_verify_deps_before_run=false
```

I think this will work unless someone explicitly sets the `verify-deps-before-run` setting via CLI options. However, in that case it is probably expected that we don't skip the check.

Reply:
Replacing it breaks some tests related to nested scripts. It probably means that it fails to skip check in some situations.

The code is in branch [replace-custom-env-with-npm-config-env](https://github.com/pnpm/pnpm/tree/replace-custom-env-with-npm-config-env). The commit is https://github.com/pnpm/pnpm/commit/907b58a4bd3b7ad0a99afcb37b05d46f5971be10.